updater/lib/dependabot/updater/operations/update_all_versions.rb

```diff
@@ -185,55 +168,11 @@ def file_updater_for(dependencies)
           )
         end
 
-        def security_advisories_for(dep)
-          relevant_advisories =
-            job.security_advisories.
-            select { |adv| adv.fetch("dependency-name").casecmp(dep.name).zero? }
-
-          relevant_advisories.map do |adv|
-            vulnerable_versions = adv["affected-versions"] || []
-            safe_versions = (adv["patched-versions"] || []) +
-                            (adv["unaffected-versions"] || [])
-
-            Dependabot::SecurityAdvisory.new(
-              dependency_name: dep.name,
-              package_manager: job.package_manager,
-              vulnerable_versions: vulnerable_versions,
-              safe_versions: safe_versions
-            )
-          end
-        end
-
         def log_checking_for_update(dependency)
           Dependabot.logger.info(
             "Checking if #{dependency.name} #{dependency.version} needs updating"
           )
-          log_ignore_conditions(dependency)
-        end
-
-        def log_ignore_conditions(dep)
-          conditions = job.ignore_conditions.
-                       select { |ic| name_match?(ic["dependency-name"], dep.name) }
-          return if conditions.empty?
-
-          Dependabot.logger.info("Ignored versions:")
-          conditions.each do |ic|
-            unless ic["version-requirement"].nil?
-              Dependabot.logger.info("  #{ic['version-requirement']} - from #{ic['source']}")
-            end
-
-            ic["update-types"]&.each do |update_type|
-              msg = "  #{update_type} - from #{ic['source']}"
-              Dependabot.logger.info(msg)
-            end
-          end
-        end
-
-        def name_match?(name1, name2)
-          WildcardMatcher.match?(
-            job.name_normaliser.call(name1),
-            job.name_normaliser.call(name2)
-          )
+          job.log_ignore_conditions_for(dependency)
```

Comment:
I see this is a refactor, where behavior shouldn't change at all. I'm still going to take the opportunity to ask about the code that's being updated 😇 

It feels 🤏 strange to me to have an operations class tell the job to log something. I wonder if in the longer term it would make sense for the job to return the ignore conditions in a format that can be logged. I have a feeling this is related to your comment in the job class, which I commented on.

Reply:
Yeah, this is definitely clunky, if it wasn't for the size of the method required to parse the 'raw' `ignore_conditions` I wouldn't mind it being duplicated.

I did also think about having a mixin that collects up all the common logging methods which might be an adequate middle ground but I punted on it as part of this as there's actually fewer shared logging methods that it seems.

My first preference would be that we effectively only have one notation of ignore conditions, as objects, available on the Job and then have a single log-presentation method but that's annoying non-trivial.